### PR TITLE
P2022-1491-Validate-Board-State-In-patchBoard

### PR DESCRIPTION
**Pull request by Radenzvous**
…s not allow modification of the maximum number of votes

### CODE REVIEW

**Comment by patrykzawadzkiintive** [comment]
Code looks good, but you need to create a unit test for that. What happened when you update number of votes when board state is created, and what happened if other state.

**Review by patrykzawadzkiintive — CHANGES_REQUESTED** [review]
Code looks good, but you need to create a unit test for that. What happened when you update number of votes when board state is created, and what happened if other state.

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 1]
```diff
@@ -76,10 +68,16 @@ public BoardDto patchBoard(final Integer id, final BoardPatchDto boardPatchDto,
...
 
-        board.map(b -> Optional.ofNullable(b.getCreator()).filter(creator -> creator.getEmail().equals(email))
-                .orElseThrow(() -> new BadRequestException("Not a board owner!")));
-
         boardReturn = board.map(b -> {
+            if (b.getState() == EnumStateDto.CREATED) {
+                Optional.ofNullable(b.getCreator())
+                        .filter(creator -> creator.getEmail().equals(email))
+                        .orElseThrow(() -> new BadRequestException
+                                ("Not a board owner!"));
```
Comment (daniel-kmiecik) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:76]: Formatting

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 2]
```diff
@@ -76,10 +68,16 @@ public BoardDto patchBoard(final Integer id, final BoardPatchDto boardPatchDto,
...
+            } else {
+                throw new BadRequestException
+                        ("State of board does not allow to change number of votes!");
```
Comment (daniel-kmiecik) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:79]: Formatting - unnecessary new line

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 3]
```diff
@@ -342,7 +338,8 @@ void patchBoardShouldReturnUserIsNotAnOwner() {
         final var email = "username@test.pl";
         final var user = new User(uid, email, "displayName", Set.of());
         final var id = 10;
-        final var board = buildBoard(user);
+        final var state = EnumStateDto.CREATED;
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:341]: you don't need to create state as separate variable if it's used only in line below
```suggestion
        final var board = buildBoard(user, EnumStateDto.CREATED);
```

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 4]
```diff
@@ -359,7 +356,8 @@ void patchBoardShouldUpdateNameAndNumberOfVotesFields() {
         final var email = "username@test.pl";
         final var boardName = "My first board.";
         final var userOwner = new User(uid, email, "displayName", Set.of());
-        final var board = buildBoard(userOwner);
+        final var state = EnumStateDto.CREATED;
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:359]: you don't need to create state as separate variable if it's used only in line below
```suggestion
        final var board = buildBoard(user, EnumStateDto.CREATED);
```

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 5]
```diff
@@ -381,7 +379,8 @@ void patchBoardShouldUpdateNumberOfVotesWithoutProvidingName() {
         final var uid = "uid101";
         final var email = "username@test.pl";
         final var user = new User(uid, email, "displayName", Set.of());
-        final var board = buildBoard(user);
+        final var state = EnumStateDto.CREATED;
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:382]: you don't need to create state as separate variable if it's used only in line below
```suggestion
        final var board = buildBoard(user, EnumStateDto.CREATED);
```

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 6]
```diff
@@ -438,10 +437,11 @@ void assignUsersToBoardShouldThrowBadRequestWhenUserIsNotOwner() {
         final String email = "testemail@example.com";
         final User user = new User("123", email, "test name", Set.of());
         final User boardOwner = new User("1234", "testemail1@example.com", "test name", Set.of());
+        final var state = EnumStateDto.CREATED;
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:440]: Same as above

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 7]
```diff
@@ -457,7 +457,8 @@ void assignUsersToBoardShouldReturnFailedEmails() throws JSONException {
         final String displayName = "testDisplayName";
         final User owner = new User("123", ownerEmail, displayName, Set.of());
         final User user = new User("1234", usersEmails.get(0), displayName, new HashSet<>());
-        final Board board = buildBoard(owner);
+        final var state = EnumStateDto.CREATED;
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:460]: you don't need to create state as separate variable if it's used only in line below
```suggestion
        final var board = buildBoard(user, EnumStateDto.CREATED);
```

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 8]
```diff
@@ -471,11 +472,50 @@ void assignUsersToBoardShouldReturnFailedEmails() throws JSONException {
...
     }
 
-    private Board buildBoard(final User user) {
+    @Test
+    void patchBoardShouldReturnBadRequestWhenBoardStateNotCreated() {
+        // given
+        final var uid = "1234";
+        final var email = "John@test.pl";
+        final var user = new User(uid, email, "john14", Set.of());
+        final var state = EnumStateDto.VOTING;
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:481]: you don't need to create state as separate variable if it's used only in line below
```suggestion
        final var board = buildBoard(user, EnumStateDto.CREATED);
```

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 9]
```diff
@@ -471,11 +472,50 @@ void assignUsersToBoardShouldReturnFailedEmails() throws JSONException {
...
+                () -> boardService.patchBoard(id, boardPatchDto, email));
+    }
+
+    @Test
+    void patchBoardShouldReturnOkWhenStateIsCreated() {
+        // given
+        final var uid = "1234";
+        final var email = "John@test.pl";
+        final var user = new User(uid, email, "john14", Set.of());
+        final var state = EnumStateDto.CREATED;
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:499]: you don't need to create state as separate variable if it's used only in line below
```suggestion
        final var board = buildBoard(user, EnumStateDto.CREATED);
```

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 10]
```diff
@@ -471,11 +472,50 @@ void assignUsersToBoardShouldReturnFailedEmails() throws JSONException {
         assertEquals(failedEmails.get(0), usersEmails.get(1));
     }
 
-    private Board buildBoard(final User user) {
+    @Test
+    void patchBoardShouldReturnBadRequestWhenBoardStateNotCreated() {
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:476]: patchBoardShouldReturnBadRequestWhenBoardState**Is**NotCreated

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 11]
```diff
@@ -76,14 +68,17 @@ public BoardDto patchBoard(final Integer id, final BoardPatchDto boardPatchDto,
...
-                .orElseThrow(() -> new BadRequestException("Not a board owner!")));
-
         boardReturn = board.map(b -> {
+            if (b.getState() == EnumStateDto.CREATED) {
+                Optional.ofNullable(b.getCreator())
+                        .filter(creator -> creator.getEmail().equals(email))
+                        .orElseThrow(() -> new BadRequestException("Not a board owner!"));
+            } else {
+                throw new BadRequestException("State of board does not allow to change number of votes!");
+            }
```
Comment (patrykzawadzkiintive) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:78]: add empty line to separate that if else from next if

**Review by patrykzawadzkiintive — CHANGES_REQUESTED** [review]

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java (hunk 1 above)
Comment (Radenzvous) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java]: done

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java (hunk 2 above)
Comment (Radenzvous) [reply src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java]: done

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java (hunk 4 above)
Comment (Radenzvous) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java]: done

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java (hunk 5 above)
Comment (Radenzvous) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java]: done

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java (hunk 6 above)
Comment (Radenzvous) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java]: done

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java (hunk 7 above)
Comment (Radenzvous) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java]: done

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java (hunk 8 above)
Comment (Radenzvous) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java]: done

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java (hunk 9 above)
Comment (Radenzvous) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java]: done

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java (hunk 10 above)
Comment (Radenzvous) [reply src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java]: done

**Review by patrykzawadzkiintive — APPROVED** [review]

**Review by daniel-kmiecik — APPROVED** [review]

**Review by edych — APPROVED** [review]

**Pull request merged**